python/ola/MACAddress.py

```diff
@@ -51,18 +51,37 @@ def __hash__(self):
   def __repr__(self):
     return self.__str__()
 
-  def __cmp__(self, other):
```

Comment:
I wonder if this class should implement `__hash__` because anyone could write code like this which would change the value of the class :

```python
address = MACAddress(bytearray([0x01, 0x23, 0x45, 0x67, 0x89, 0xab]))  # creates a MACAddress
mac = address.mac_address  # gets its representation inside the class
print(hash(address))  # print the class' hash value, for example 1043151800894266584
mac[0] = 0xff  # changes the MAC address, shouldn't be possible but it is.
print(hash(address))  # prints the class' hash value, different from the first one, for example 6608185319733256019
```
And would get different results at the two print statements, without violating any python (implicit) rules.

Reply:
Yep, this agrees it's bad:
https://stackoverflow.com/questions/13041352/python-quickly-hash-mutable-object

Going id based would solve our issue wouldn't it? Even if someone fiddles with the value, it's still the same object? Probably not ideal, but at least safer.

Reply:
We should still try and fix this please. It looks like a Python module just hashes the value:
https://github.com/drkjam/netaddr/blob/rel-0.7.x/netaddr/eui/__init__.py#L550-L552

But I'd agree we should do something or switch to "id".